Packages/ClientRuntime/Sources/PrimitiveTypeExtensions/String+Extensions.swift

```diff
@@ -113,3 +113,10 @@ extension String {
         return ByteBuffer(data: self.data(using: .utf8) ?? Data())
     }
 }
+
+public extension Optional where Wrapped: Collection {
+  var isNilOrEmpty: Bool {
+    guard let value = self else { return true }
+    return value.isEmpty
+  }
+}
```

Comment:
1.  Does this need to be public?  I guess it does since a package like AWSGlacier wants to be able to use it?
2. It looks like this extension is applied to all optional collections?  If this is true, is this what we want?  If this is what we want, seems like this should be in a separate file (as it is currently in a file called `String+Extensions`
3. I don't understand how `where Wrapped` works -- can you explain?

Reply:
removing per changes in other PR